evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -870,8 +874,23 @@ def test_compute_estimator_features_nonlinear(
     mock_ohe.return_value = pd.DataFrame(X)
     mock_en_predict.return_value = pd.Series(np.ones(X.shape[0]))
     mock_rf_predict.return_value = pd.Series(np.zeros(X.shape[0]))
+
+    mock_en_predict_proba_df = pd.DataFrame(
+        {0: np.zeros(X.shape[0]), 1: np.ones(X.shape[0])}
+    )
+    mock_en_predict_proba_df.ww.init()
+    mock_en_predict_proba.return_value = mock_en_predict_proba_df
+    mock_rf_predict_proba_df = pd.DataFrame(
+        {0: np.ones(X.shape[0]), 1: np.zeros(X.shape[0])}
+    )
+    mock_rf_predict_proba_df.ww.init()
+    mock_rf_predict_proba.return_value = mock_rf_predict_proba_df
```

Comment:
This is the same as `mock_en_predict_proba_df`, right? Do we need to initialize two separate dfs or can we just use the same one?

Reply:
They're slightly different, `en` is a column of all zeros and then a column of all ones, whereas `rf` is a column of all ones then a column of all zeros. It's more to make sure that we're choosing the correct columns and appending values correctly.